### ISSUE TITLE
[question] Possible to reduce disk usage when building a dependency tree?

### ISSUE THREAD

**Issue by schaber-hxg** [conan-io/conan#17460]
### What is your question?

Hi! I have a question if conan has an option to save disk space while building multiple packages with `conan create` using `--build=missing`. 

Context:
We use conan v2.7 (upgrade is possible) to manage our third party dependencies. We have a CI pipeline that allows us to build all dependencies of our project at once. Due to the amount of packages and some large ones the requirement on disk space is > 150GB for the local cache during the build.

Our question is if it is possible to clean the build artifacts after each build of a single package, as these should no longer be required.

**Comment by jcar87** [comment]
Hi @schaber-hxg - thanks for asking this question

> Our question is if it is possible to clean the build artifacts after each build of a single package, as these should no longer be required.

This would be the purpose of the `conan cache clean` command, docs here: 
https://docs.conan.io/2/reference/commands/cache.html#conan-cache-clean

by default it would remove all "temp"/"intermediate" files like temporary build folders, etc - and retain only the package folders (these tend to be small for most libraries). I'd imagine the biggest contributor to storage are the build folders (with the temporary .o/obj files before linking, etc)

**Comment by schaber-hxg** [comment]
Thanks for the answer, I know about this command, however it can't help us in our case. We build all dependencies with `conan create [...] --build=missing`, so this is a single command and we cannot call `conan cache clean` while it runs. We can clean disk space after building all packages, but have been wondering if it is possible to remove the build folder **while** running `conan create`. I know this is a very special and unusual request, and I assume it's not possible but still wanted to ask.

**Comment by jcar87** [comment]
Thanks @schaber-hxg  - will transfer this to the Conan issue tracker (rather than Conan Center) - as this falls in line within best CI practices which we are currently working on :)

**Comment by memsharded** [comment]
Hi @schaber-hxg 

Thanks for the feedback. 

This kind of very large builds are typically addressed distributedly, using the ``conan graph build-order`` that allows to send build work loads in parallel to different build agents. Since last week there is a new complete section in the docs called CI-Tutorial in https://docs.conan.io/2/ci_tutorial/tutorial.html that explains how this can be done. Even if the build is not distributed, this might be executed in the same machine, it will require a bit more effort (like iterating the "build-order"),  just a few extra lines in the CI.

Otherwise, have you considered maybe using some ``post_package()`` hook? I think after the post-package you might be able to remove the contents of the ``build_folder`` directly from the hook.

**Comment by schaber-hxg** [comment]
We had something like using the build-order as an alternative in mind. Thank you for these directions!
Will close the issue as we have options to work with, thank you.

**Comment by memsharded** [comment]
Great!
If you have any further issue or question, don't hesitate to re-open or create a new ticket. Thanks for the feedback!
